### PR TITLE
fix: QnA import sign in

### PR DESCRIPTION
**Pull request by zhixzhan**
## Description
- Fix Sign in issue.
- Update text.

## Task Item
#minor

## Screenshots
image
test1

### CODE REVIEW

**Comment by srinaath** [comment]
@zhixzhan It would be great to capture "All" references to the word `KB` in the codebase to `Knowledge Base` as a part of this PR.

File: Composer/packages/client/src/components/QnA/CreateQnAFromQnAMaker.tsx [hunk 1]
```diff
@@ -33,6 +37,32 @@ export const CreateQnAFromQnAMaker: React.FC<CreateQnAFromFormProps> = (props) =
     onChange(formData, disabled);
   }, [formData]);
 
+  useEffect(() => {
+    const shouldProvideTokens = userShouldProvideTokens();
+    if (!shouldProvideTokens) {
+      AuthClient.getAccount().then((account) => {
```
Comment (hatpick) [inline Composer/packages/client/src/components/QnA/CreateQnAFromQnAMaker.tsx:43]: use await

File: Composer/packages/client/src/components/QnA/CreateQnAFromQnAMaker.tsx [hunk 2]
```diff
@@ -33,6 +37,32 @@ export const CreateQnAFromQnAMaker: React.FC<CreateQnAFromFormProps> = (props) =
...
+        }
+      }
+    }
+  }, []);
+
+  const decodeToken = (token: string) => {
+    try {
+      return jwtDecode<any>(token);
+    } catch (err) {
+      console.error('decode token error in ', err);
```
Comment (hatpick) [inline Composer/packages/client/src/components/QnA/CreateQnAFromQnAMaker.tsx:61]: avoid logging to console

File: Composer/packages/client/src/components/QnA/CreateQnAFromQnAMaker.tsx [hunk 3]
```diff
@@ -47,9 +77,20 @@ export const CreateQnAFromQnAMaker: React.FC<CreateQnAFromFormProps> = (props) =
           </span>
         </p>
       </Stack>
-      <Stack>
-        <Link>{formatMessage('Login to Azure to continue')}</Link>
-      </Stack>
+
+      {signedInAccount ? (
+        <div style={accountInfo}>
+          <span>{`Signed in as ${signedInAccount}. Click `}</span>
```
Comment (hatpick) [inline Composer/packages/client/src/components/QnA/CreateQnAFromQnAMaker.tsx:83]: formatMessage

File: Composer/packages/client/src/components/QnA/CreateQnAFromQnAMaker.tsx [hunk 4]
```diff
@@ -22,7 +25,8 @@ const formConfig: FieldConfig<CreateQnAFromQnAMakerFormData> = {
 };
 
 export const CreateQnAFromQnAMaker: React.FC<CreateQnAFromFormProps> = (props) => {
-  const { onChange, qnaFiles, initialName } = props;
+  const { onChange, qnaFiles, initialName, onNext } = props;
+  const [signedInAccount, setSignedInAccount] = useState('');
```
Comment (hatpick) [inline Composer/packages/client/src/components/QnA/CreateQnAFromQnAMaker.tsx:29]: this is not an account, it's a string showing the username/email right? my suggestion: accountEmail

File: Composer/packages/client/src/components/QnA/CreateQnAFromQnAMaker.tsx [hunk 5]
```diff
@@ -47,9 +77,20 @@ export const CreateQnAFromQnAMaker: React.FC<CreateQnAFromFormProps> = (props) =
...
       </Stack>
-      <Stack>
-        <Link>{formatMessage('Login to Azure to continue')}</Link>
-      </Stack>
+
+      {signedInAccount ? (
+        <div style={accountInfo}>
+          <span>{`Signed in as ${signedInAccount}. Click `}</span>
+          <span style={signInButton} onClick={onNext}>
+            {'next '}
```
Comment (hatpick) [inline Composer/packages/client/src/components/QnA/CreateQnAFromQnAMaker.tsx:85]: formatMessage

File: Composer/packages/client/src/components/QnA/CreateQnAFromQnAMaker.tsx [hunk 6]
```diff
@@ -47,9 +77,20 @@ export const CreateQnAFromQnAMaker: React.FC<CreateQnAFromFormProps> = (props) =
...
+          </span>
+          <span>{'to select KBs'}</span>
```
Comment (hatpick) [inline Composer/packages/client/src/components/QnA/CreateQnAFromQnAMaker.tsx:87]: formatMessage and KB => knowledge base

File: Composer/packages/client/src/components/QnA/CreateQnAFromQnAMaker.tsx [hunk 7]
```diff
@@ -47,9 +77,20 @@ export const CreateQnAFromQnAMaker: React.FC<CreateQnAFromFormProps> = (props) =
...
+        </div>
+      ) : (
+        <div style={signInButton} onClick={onNext}>
+          {formatMessage('Sign in to Azure to continue')}
```
Comment (hatpick) [inline Composer/packages/client/src/components/QnA/CreateQnAFromQnAMaker.tsx:91]: formatMessage

File: Composer/packages/client/src/components/QnA/ReplaceQnAFromModal.tsx [hunk 8]
```diff
@@ -231,6 +231,7 @@ export const ReplaceQnAFromModal: React.FC<ReplaceQnAModalProps> = (props) => {
         if (decoded) {
           setToken(newtoken);
           setUserProvidedTokens(true);
+          setSignedInAccount(decoded.email);
```
Comment (hatpick) [inline Composer/packages/client/src/components/QnA/ReplaceQnAFromModal.tsx:234]: same comment

File: Composer/packages/client/src/components/QnA/ReplaceQnAFromModal.tsx [hunk 9]
```diff
@@ -395,20 +396,19 @@ export const ReplaceQnAFromModal: React.FC<ReplaceQnAModalProps> = (props) => {
...
-                        {'next '}
-                      </span>
-                      <span>{'to select KBs'}</span>
-                    </div>
-                  ) : (
-                    <div style={signInButton} onClick={performNextAction}>
-                      {formatMessage('Sign in to Azure to continue')}
-                    </div>
-                  ))}
+                {signedInAccount ? (
```
Comment (hatpick) [inline Composer/packages/client/src/components/QnA/ReplaceQnAFromModal.tsx:399]: maybe wrap this piece of UI into it's own component and reuse it (+ same comments as above)

**Review by hatpick — CHANGES_REQUESTED** [review]

**Comment by zhixzhan** [comment]
Align to global auth PR https://github.com/microsoft/BotFramework-Composer/pull/8291
Closing this.

**Pull request closed without merging**